### ISSUE TITLE
WANT Compose API URL config

### ISSUE THREAD

**Issue by Toasterson** [cortezaproject/corteza-server-corredor#5]
Corredor as built with the docker container uses api.$ContainerHash as Compose API Host which ofcourse does not work as the container only contains corredor and not the API.

I have not found a documented way to tell corredor to use the correct corteza Hostname.

For reference this is the error in the log file
```json
{"level":"warn","ts":1595872426.2998362,"logger":"corredor","caller":"corredor/service.go:793","msg":"corredor responded with error","script":"/server-scripts/crm/Contact/SetLabel.js:default","runAs":"","args":"beforeUpdate","resource":"compose:record","invoker":"154290662374640170","error":"rpc error: code = Unknown desc = getaddrinfo ENOTFOUND api.148188cb449a"}
```

**Comment by darh** [comment]
Apologies for not replying sooner. Is this still an issue? Not sure what you mean by `api.$ContainerHash `? There is no such thing in Corteza or Corredor.

You can see how config is parsed here:
https://github.com/cortezaproject/corteza-server-corredor/blob/2021.9.x/src/config.ts

Or in the docs:
https://docs.cortezaproject.org/corteza-docs/2021.3/devops-guide/configuration/corredor.html

**Comment by Toasterson** [comment]
Hi @darh Yes this is still a problem. As the discover function inside a container returns the containers hostname not the domain name from a normal linux system as it expects. Thus if corredor and corteza are in seperate containers corredor cannot start.

**Comment by darh** [comment]
I understand. Please see the docs (and maybe the config.ts). Let me know if you've managed to resolve it.

**Comment by tjerman** [comment]
Newer Corteza versions, when used with docker-compose, should work out of the box as long as the `.env` variables are properly set (mainly the `DOMAIN` one).

In cases where Corredor can't correctly figure out where things are (the latest version should display an error on startup if it can't reach the Corteza API), you can use these two .env variables to control it:

```
CORREDOR_EXEC_CSERVERS_API_HOST
CORREDOR_EXEC_CSERVERS_API_BASEURL_TEMPLATE
```

If you're running the API and web applications in separate containers (you have different domain sub-domains) your variables might look something like this:

```
CORREDOR_EXEC_CSERVERS_API_HOST=api.my-corteza-here.tld
CORREDOR_EXEC_CSERVERS_API_BASEURL_TEMPLATE=https://{host}/{service}
```

If you're running the API and web applications in the same container (you have the same sub-domain) your variables might look something like this:

```
CORREDOR_EXEC_CSERVERS_API_HOST=my-corteza-here.tld
CORREDOR_EXEC_CSERVERS_API_BASEURL_TEMPLATE=https://{host}/api/{service}
```

(notice the difference in the `api` bit).

We will include this in the documentation in the next round of revisions.

You could also consider switching from automation scripts to [workflows](https://docs.cortezaproject.org/corteza-docs/2021.9/integrator-guide/automation/workflows/index.html) -- the new automation system released back in version `2021.3`
